server/src/main/java/org/elasticsearch/index/query/InnerHitContextBuilder.java

```diff
@@ -114,6 +115,16 @@ protected void setupInnerHitsContext(
         if (innerHitBuilder.getSorts() != null) {
             Optional<SortAndFormats> optionalSort = SortBuilder.buildSort(innerHitBuilder.getSorts(), searchExecutionContext);
             if (optionalSort.isPresent()) {
+                if (innerHitBuilder.getRescoreBuilders() != null) {
```

Comment:
We should have a test for this

Reply:
Why we can't not use sort with rescores?  I assumed we use `sort` to sort all inner hits and rescorer to rescore `window_size` of them after sort is applied.